piecrust/src/vm.rs

```diff
@@ -248,5 +256,18 @@ impl HostQueries {
 /// function, and should be processed first. Once this is done, the implementor
 /// should emplace the return of the query in the same buffer, and return the
 /// length written.
-pub trait HostQuery: Send + Sync + Fn(&mut [u8], u32) -> u32 {}
-impl<F> HostQuery for F where F: Send + Sync + Fn(&mut [u8], u32) -> u32 {}
+///
+/// The host query will have access to the underlying session, and can use it to
+/// perform calls to other contracts.
+///
+/// # Panics
+/// The implementor should panic if any error occurs during the execution of a
```

Comment:
I think you need to find a different phrasing for "The implementor should panic" ;)